### PR TITLE
Ccip-3076 deployment package

### PR DESCRIPTION
**Pull request by AnieeG**
This exposes the new style of deployment/configuration logic for other products to use. The general structure and usage is described in the README. 
Notes
- ~5k is generated protobuf code to unblock. That will be imported once exposed.
- ~5k is generated graphql code for programmatic CL node interaction (integration-tests/web/sdk)
- We put the deployment/configuration logic in integration-tests module for a few reasons:
	- Keeps the chain dependencies out of the core module, in particular helpful for eventual cross family tests
	- It can become the canonical deployment logic to be used for CRIB envs as well (eventually can replace the actions + contracts dirs)
- To accomplish the lightweight tests (chainlink.Application + simulated.Backend) we expose some test utilities in util/testutils/

### CODE REVIEW

**Comment by lukaszcl** [comment]
Merged develop to fix the CI builds.

File: integration-tests/deployment/memory/environment.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		Name:     Memory,
+		Offchain: NewMemoryJobClient(nodes),
+		// Note these have the p2p_ prefix.
+		NodeIDs: nodeIDs,
+		Chains:  chains,
+		Logger:  lggr,
+	}
+}
+
+//func NewMemoryEnvironmentExistingChains(t *testing.T, lggr logger.Logger,
```
Comment (patrickhuie19) [inline integration-tests/deployment/memory/environment.go:107]: /nit remove?

File: integration-tests/deployment/memory/job_client.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	Nodes map[string]Node
+}
+
+func (j JobClient) GetNode(ctx context.Context, in *nodev1.GetNodeRequest, opts ...grpc.CallOption) (*nodev1.GetNodeResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (j JobClient) ListNodes(ctx context.Context, in *nodev1.ListNodesRequest, opts ...grpc.CallOption) (*nodev1.ListNodesResponse, error) {
+	//TODO implement me
```
Comment (patrickhuie19) [inline integration-tests/deployment/memory/job_client.go]: /nit ticket?

File: integration-tests/web/sdk/client/client.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	"net/http"
+	"strings"
+
+	"github.com/Khan/genqlient/graphql"
+
+	"github.com/smartcontractkit/chainlink/integration-tests/web/sdk/client/internal/doer"
+	"github.com/smartcontractkit/chainlink/integration-tests/web/sdk/internal/generated"
+)
+
+type Client interface {
```
Comment (patrickhuie19) [inline integration-tests/web/sdk/client/client.go:16]: We have similar web routes defined in `core/web` - is there an ability to consolidate/reuse existing implementations?

File: integration-tests/deployment/memory/environment.go (hunk 1 above)
Comment (AnieeG) [reply integration-tests/deployment/memory/environment.go:107]: @connorwstein do we need to keep it?

Comment (connorwstein) [reply integration-tests/deployment/memory/environment.go:107]: can remove

File: integration-tests/web/sdk/client/client.go (hunk 3 above)
Comment (MStreet3) [reply integration-tests/web/sdk/client/client.go:16]: can you link to what implementations you're referring to?

File: integration-tests/deployment/environment.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+func (e Environment) AllChainSelectors() []uint64 {
+	var selectors []uint64
+	for sel := range e.Chains {
+		selectors = append(selectors, sel)
+	}
+	return selectors
+}
+
+func ConfirmIfNoError(chain Chain, tx *types.Transaction, err error) (uint64, error) {
```
Comment (AnieeG) [inline integration-tests/deployment/environment.go:62]: @connorwstein changed the signature of this as discussed

File: integration-tests/deployment/environment.go [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+	nodev1.NodeServiceClient
+}
+
+type Chain struct {
+	// Selectors used as canonical chain identifier.
+	Selector uint64
+	Client   OnchainClient
+	// Note the Sign function can be abstract supporting a variety of key storage mechanisms (e.g. KMS etc).
+	DeployerKey *bind.TransactOpts
+	Confirm     func(tx common.Hash) (uint64, error)
```
Comment (AnieeG) [inline integration-tests/deployment/environment.go:43]: @connorwstein changed the signature of this as discussed

File: integration-tests/web/sdk/client/client.go (hunk 3 above)
Comment (patrickhuie19) [reply integration-tests/web/sdk/client/client.go:16]: hmm, my bad. I thought we had defined go APIs, but I was thinking of the scattered resolver unit tests we have that use graphql queries. 

for instance, https://github.com/smartcontractkit/chainlink/blob/677abe19d1fc15e6fa248271eab0aacb2ae4c0b7/core/web/resolver/job_proposal_test.go#L20

**Review by patrickhuie19 — DISMISSED** [review]

**Review by skudasov — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

**Review by patrickhuie19 — APPROVED** [review]

**Pull request merged**